README.md

```diff
@@ -52,17 +30,13 @@ In order to be able to use this application while running the tool locally, make
 
 3. Retrieve the values for your application's Client ID and Client Secret from the [GitHub Developer Settings page](https://github.com/settings/developers) and add them to the end of the corresponding lines in .env.
 
-## Installation
+### Docker
 
-You need to satisfy all [project requirements](#requirements), and then run:
+docker run --env-file .env --rm -p 8080:8080 srcd/code-annotation
```

Comment:
I find extremely odd having a `.env` file and not very devops friendly. If we are going ot have it, at least include the definition of each variable in the templete. But I prefer having it explained in the readme and  having defaults.

Reply:
I think the `.env` file is just an easy way to have in one place all the environmental values as expected by the app, but it is up to the developers to decide how are they providing them; it can be passed as regular vars (ie.`UI_DOMAIN=http://127.0.0.1:3000 make gorun`) or as regular docker envs (ie. `docker run --env OAUTH_CLIENT_ID=id --env OAUTH_CLIENT_SECRET=secret --rm -p 8080:8080 srcd/code-annotation`)

Most of them are documented in different parts of the `README.md` or the `CONTRIBUTING.md`.

@mcuadros Should we block this PR till everything is properly documented, or can we merge and handle this in a separated issue?

These are the environment values recognized by the app; 
most of them are not mandatory, but we can also make "optional" the `JWT_SIGNING_KEY`.

key | required | actual default | default | meaning
-|-|-|-|-
`OAUTH_CLIENT_ID` | :exclamation: | | | OAuth app client id
`OAUTH_CLIENT_SECRET` | :exclamation: | | | OAuth app client secret
`JWT_SIGNING_KEY` | :exclamation: | | `jwt_signing_key` | key to sign the user session JWT
`DB_CONNECTION` | | `sqlite://internal.db` | `sqlite:///var/code-annotation/internal.db` | DSN of the internal database
`OAUTH_RESTRICT_ACCESS` | | | | group or team of users that can access the app
`OAUTH_RESTRICT_REQUESTER_ACCESS` | | | | group or team of users granted as Requesters
`REACT_APP_SERVER_URL` | | `127.0.0.1:8080` | `//code-annotation-staging.srcd.run` | `host:port` of the API
`UI_DOMAIN` | | `$(REACT_APP_SERVER_URL)` | `$(REACT_APP_SERVER_URL)` | `host:port` of the frontend recieveing the user token after authentication

Reply:
`.env` file is just a simple way to see/edit environmental variables for local development. Most of the values have defaults. All variables must and will be documented. But this PR isn't the right place to do that.

About devops:
In my experience env vars are the most friendly and natural way to configure an application with kubernetes, thanks to `configmap`,`secrets` objects and `valueFrom` in container spec.

And [The twelve-factor app](https://12factor.net) also [recommends](https://12factor.net/config) it.

Reply:
@dpordomingo thank you for great summary! 

As on @mcuadros notice, we should have it clearly documented - have added it to the scope of #67

Reply:
@smacker where says that is recommended have a `.env` file? This is not recommended at all, and almost none use this.

Also is bad not having any prefix on your env vars.

If you are speaking about the usage of env vars, of course, is recommended, and actually is a MUST that all source{d} applications

Reply:
@mcuadros no-no. I mean env vars, not `.env` file. We use `.env` file ONLY for local development. Agreed about prefix, we should add it to env vars without prefix! Thanks!

Didn't know it's must, https://github.com/bblfsh/dashboard uses flags instead. But glad to hear.